### PR TITLE
Adding unsupported text into the team/e0 edition menu

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
Changing the text for the free/e0 edition to show in the product menu a text that clarifies that the edition is unsupported.

#### Ticket Link
Not Yet

#### Screenshots
Before:
imagen

After:
image
#### Release Note

```release-note
Added unsupported label to the Team/E0 editions in the product menu.
```

### CODE REVIEW

**Review by matthewbirtch — APPROVED** [review]
Approving this with one _really_ minor change @jespino. 

The 'unsupported' link seems to have a background color on hover. Can you adjust so it only has the underline on hover?

image

File: webapp/channels/src/components/widgets/menu/menu_items/menu_start_trial.tsx [hunk 1]
```diff
@@ -84,20 +85,24 @@ const MenuStartTrial = (props: Props): JSX.Element | null => {
...
+            <div className='editionText'>
+                {formatMessage(
+                    {
+                        id: 'navbar_dropdown.versionText',
+                        defaultMessage: 'This is the free <link>unsupported</link> edition of Mattermost.',
+                    },
+                    {
+                        link: (msg: React.ReactNode) => (
+                            <ExternalLink
+                                location='unsupported-link'
```
Comment (nickmisasi) [inline webapp/channels/src/components/widgets/menu/menu_items/menu_start_trial.tsx:97]: I generally try to be more specific with the location so we can determine where exactly this was hit - something like `menu_start_trial.unsupported-link`

**Review by nickmisasi — APPROVED** [review]
One small suggestion otherwise LGTM!

File: webapp/channels/src/components/widgets/menu/menu_items/menu_start_trial.tsx (hunk 1 above)
Comment (jespino) [reply webapp/channels/src/components/widgets/menu/menu_items/menu_start_trial.tsx]: Done

**Comment by jespino** [comment]
@matthewbirtch I made the changes on the styles addressing your proposal

**Review by matthewbirtch — APPROVED** [review]

**Review by lindy65 — APPROVED** [review]
Thanks @jespino - looks good to me 👍 

Please link to the Jira ticket? Thanks!

**Comment by hmhealey** [comment]
What does "unsupported" mean in this context? I assume it's supposed to mean that we won't be providing official support to people, but based on the link, that's not obvious to me. That wording really makes it sound like we're not doing releases or maintaining TE going forward

**Comment by matthewbirtch** [comment]
> What does "unsupported" mean in this context? I assume it's supposed to mean that we won't be providing official support to people, but based on the link, that's not obvious to me. That wording really makes it sound like we're not doing releases or maintaining TE going forward

@hmhealey this text was mulled over with leadership last week and this is where it landed. I understand what you're saying though. I had a similar reaction to the word at first. "Unsupported" here means that commercial support is not included in the free unlicensed versions.

cc: @chenilim for thoughts

**Comment by matthewbirtch** [comment]
@jespino @nickmisasi how we verify that the link tracking is working as expected? Just want to be sure we get data coming in properly.

**Comment by nickmisasi** [comment]
@matthewbirtch click the link you're looking to validate, and then inspect the query parameters of the URL you're redirected to. For example when I click the link in the spinwick I see:

https://mattermost.com/pricing/?utm_source=&utm_medium%20mattermost%20=in-product&utm_content=menu_start_trial.unsupported-link&uid=h9ciej8xk3gq5xps5hh3nn5eyh&sid=nqs437yx6j8idynxsiahbd3pgw
the utm parameters and `uid/sid` are what's added

**Comment by matthewbirtch** [comment]
> @matthewbirtch click the link you're looking to validate, and then inspect the query parameters of the URL you're redirected to.

Thanks @nickmisasi. To clarify, the tracking data we get from this will be in Google Analytics right? Is there some validation we can do there to ensure it's getting tracked correctly (other than seeing the parameters in the query string)

**Comment by nickmisasi** [comment]
@matthewbirtch the tracking data will be _available_ to Google Analytics - but whether/how it's consumed by GA is owned by the marketing team. Generally GA captures all query parameters automatically, but to totally validate you'd need to work with them. 

FWIW, this tracking data is also emitted through rudder at the time of clicking the link.

**Comment by it33** [comment]
@matthewbirtch @jespino can we delete "Start an Enterprise trial"? That was discussed, debated and deleted.

**Comment by matthewbirtch** [comment]
Thanks @it33, we will definitely remove this. I should have caught this. 

@jespino let me know if you have any questions, but here is some extra context to share for those following along: 

1. A 'start free trial' call-to-action in the chat-facing side of the application is not aligned with our mission-critical brand. It is more akin to methods SaaS products use in SMB markets. It's like the 'upsells' we've used in the past and need to move away from.
2. The 'unsupported' message is intended to communicate the seriousness of running unsupported software in a mission-critical organization. However, the "Start free trial" language is at odds with that. It downplays the importance of it and cheapens it.
3. Another concern here is that admins see this 'free trial' button and because it appears on the chat-facing side, they might worry that all end users see the same thing. We don't want to add this stress to our admins.

Happy to have a broader conversation about this and we can chat more in our team meetings.

**Review by hmhealey — APPROVED** [review]
From a code perspective, this looks good to me. I still think the wording will scare people more than we should, but that's not my call to make

**Comment by hamidout** [comment]
is there anyway to hide this message from showing up for users via the admin console or config file?

**Comment by Zlitus** [comment]
> is there anyway to hide this message from showing up for users via the admin console or config file?

In your HTTP server, you can send a static JSON in place of the `/api/v4/license/client` endpoint.

Here an example of how to do it with Nginx:
```
server {
	server_name mm.mywebsite.com;

	location /api/v4/endpoint {
		default_type application/json;
		return 200 '{"Announcement": "true","Company": "XXX","EmailNotificationContents": "true","IsLicensed": "true","SkuShortName": "enterprise","Users": "200000"}';
	}

	# Other server configuration...
}
```

Here an example of how to do it with Caddy2WebServer:
```
mm.mywebsite.com {
	route /api/v4/license/client {
		header Content-Type application/json
		respond `{"Announcement": "true","Company": "XXX","EmailNotificationContents": "true","IsLicensed": "true","SkuShortName": "enterprise","Users": "200000"}` 200
	}

	# Other server configuration...
}
```

By doing that, the front of mattermost will think you are on a licensed version and you will not have:

Mattermost advertising for your user.
"Free version" badge and texts.
You can probably configure that on your WAF aswell (Cloudflare, Fastly, …)

**Pull request merged**